### DISCUSSION TITLE
quasar dev capacitor iOS (M1) not working

### DISCUSSION THREAD

**Discussion by xferion** [quasarframework/quasar#9037] [CLI - Capacitor / Cordova modes] [answered]
I'm trying to run a Quasar app on iOS on Mac with M1 processor. I'm using Rosetta as Quasar (or actually Webpack I think) does not work on the native M1 (or at least I could not get it to work). 

The same app does work on Android (but from my Windows machine). Also quasar build does work, but quasar dev does not. I can also run quasar dev for a web app without problems on the same Mac.

This is what I get:
```
stephan@Stephans-MacBook-Air klien-ict-app % quasar dev -capacitor -T ios 

 Dev mode.......... spa
 Pkg quasar........ v1.14.3
 Pkg @quasar/app... v2.1.6
 Debugging......... enabled

 Configured browser support (at least 87.82% of global marketshare):
 · Chrome for Android >= 90
 · Firefox for Android >= 87
 · Android >= 90
 · Chrome >= 80
 · Edge >= 87
 · Firefox >= 78
 · iOS >= 10.3
 · Opera >= 71
 · Safari >= 11

 App · Reading quasar.conf.js
 App · Checking listening address availability (192.168.1.14:true)...

 App · ⚠️  Unknown network error occurred

[  Error: TypeError [ERR_INVALID_OPT_VALUE] [ERR_INVALID_OPT_VALUE]: The value "{ port: true, host: '192.168.1  .14' }" is invalid for option "options"

  - net.js:58 
    [klien-ict-app]/[@quasar]/app/lib/helpers/net.js:58:8

  - new Promise
  
  - net.js:36 Object.module.exports.isPortAvailable
    [klien-ict-app]/[@quasar]/app/lib/helpers/net.js:36:10

  - net.js:25 module.exports.findClosestOpenPort
    [klien-ict-app]/[@quasar]/app/lib/helpers/net.js:25:30

  - quasar-dev:101 Object.parseAddress [as onAddress]
    [klien-ict-app]/[@quasar]/app/bin/quasar-dev:101:28

  - quasar-conf-file.js:277 QuasarConfFile.prepare
    [klien-ict-app]/[@quasar]/app/lib/quasar-conf-file.js:277:29

  - task_queues.js:93 processTicksAndRejections
    internal/process/task_queues.js:93:5

  - quasar-dev:215 async goLive
    [klien-ict-app]/[@quasar]/app/bin/quasar-dev:215:5
] {

  code: 'ERR_INVALID_OPT_VALUE'
}
```
It seems it is using true as port number, but this is not what is in de quasar.conf. The port is correct there and on Windows/Android the same config does work. 

Any ideas to get this working? Thanks in advance.

**Comment by rstoenescu** [comment, marked as answer]
It sure feels like quasar.conf.js > devServer > port is set to Boolean `true`. Just did a test and it outputs the same thing. Can you offer a barebones reproduction repo pls? I don't own an M1 to test this.

**Reply by xferion** [reply]
Thank you for looking into this.

Yes, it does look like that, I actually thought the same thing, but it is not the case. This is what is in quasar.conf.js:

```
 // Full list of options: https://quasar.dev/quasar-cli/quasar-conf-js#Property%3A-devServer
    devServer: {
      https: false,
      http2: false,
      host: '192.168.1.14',
      port: 8081,
      open: true,
    },
```

As for barebone reproduction, this already happens when I create a new project and only add Capacitor:

```
quasar create test-app
cd test-app
quasar mode add capacitor
quasar dev -capacitor -T ios 
```

Results in the same error:

```
 App · Checking listening address availability (0.0.0.0:true)...

 App · ⚠️  Unknown network error occurred

[  Error: TypeError [ERR_INVALID_OPT_VALUE] [ERR_INVALID_OPT_VALUE]: The value "{ port: true, host: '0.0.0.0'   }" is invalid for option "options"
```

from quasar.conf:

```
devServer: {
      https: false,
      port: 8080,
      open: true // opens browser window automatically
    },
```

**Reply by hawkeye64** [reply]
What happens if you try to specify the port from the command line?

`quasar dev -capacitor -T ios -p 8090`

**Reply by xferion** [reply]
Yes, this works thanks! However, it opens in a browser window and not in Xcode / simulator. I would expect this, as it works like that with Android. Or does it work differently for Apple? (this is the first time I use Apple with Quasar).

**Reply by hawkeye64** [reply]
I don't know as I have never done web/capacitor dev on a Mac. But, what I am trying to do is narrow down the possibilities so @rstoenescu can pinpoint any code that may be causing this, instead of hunting for it.

**Reply by xferion** [reply]
@hawkeye64 Ok, thanks for your help!

**Reply by rstoenescu** [reply]
Whatever the problem is, it's pretty easy to fix when you have an M1 chip. But I don't (at least not yet or in the following months). Can you contact me on Discord pls so we can do a remote session to investigate directly on your laptop?

**Reply by hawkeye64** [reply]
@xferion ok, figured something out.
don't do:
```bash
quasar dev -capacitor -T ios
```
It's supposed to be:
```bash
quasar dev capacitor -T ios
```
no dash (`-`) in front of capacitor.

Did you also run `quasar mode add capacitor`?

And, did you set up Android Studio as per these instructions? https://next.quasar.dev/quasar-cli/developing-capacitor-apps/preparation#ios-setup

**Reply by xferion** [reply]
That indeed resolves the port problem and specifying the port with -p is not needed. The problem with opening in the web browser instead of Xcode remains though.

Yes, I did run 'quasar mode add capacitor'. 

Would Android studio be needed to run an iOS app? I understood it would run with Xcode, which is installed. I thought Android Studio would only be needed for Android apps. I can already create and test the Android app on my Windows machine, so I did not install it on the Mac.

**Reply by hawkeye64** [reply]
@xferion Did you follow the instructions here about accepting xcode license? https://next.quasar.dev/quasar-cli/developing-capacitor-apps/preparation#ios-setup

**Reply by xferion** [reply]
Yes, I did accept the license. Thanks for your input.

The correct command for quasar dev is actually:  `quasar dev -m capacitor -T ios`
I don't know where I copied it from incorrectly, but the -m part was missing. I noticed it by looking up the command I use to start the android app on my Windows machine.

Now I get this error: 

```
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

Xcode is just installed from the app store in the default location. Or might the problem be that Quasar is running in Rosette (Intel emulator) and I believe Xcode is a native app?

**Reply by hawkeye64** [reply]
@xferion Maybe try this: https://github.com/nodejs/node-gyp/issues/569
I don't think your issues have been m1 related at all

**Reply by xferion** [reply]
@hawkeye64 You are probably right about that. It was not easy to get the app to work though. 

This was the command that was needed if someone later finds this thread:
```
sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
```

This does not work:
```
xcode-select --install # Install Command Line Tools if you haven't already.
sudo xcode-select --switch /Library/Developer/CommandLineTools # Enable command line tools
```
